packages/core/src/components/TextField/TextField.example.jsx

```diff
@@ -11,7 +11,14 @@ ReactDOM.render(
       name="single_example"
       requirementLabel="Optional"
     />
-    <TextField defaultValue="123" label="Number field" name="number_example" type="number" />
+    <TextField
+      defaultValue="123"
+      label="Number field"
+      name="number_example"
+      type="text"
+      inputmode="numeric"
+      pattern="[0-9]*"
+    />
```

Comment:
I just tested this on mobile Chrome, and it works like a charm.

Reply:
I like the idea of including additional characters based on the masked example

Reply:
how did you test pattern? doesnt it only work inside a form and with a submit button? Im wondering if this will affect teams like app3 not using native validation...

Reply:
I didn't test the pattern, but inputmode=numeric worked for bringing up a numpad on my phone